part1.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+public function postAction()
+{
+//avoid notices warnings

```

Comment:
we need to use the try and catch in action  and also need to check if $post has data then we should continue executing code. 
if ($post){
            try {
}
catch{
}
}